### ISSUE TITLE
replace-deps/paths in aliases stopped working around 1.12.3

### ISSUE THREAD

**Issue by imrekoszo** [cursive-ide/cursive#2689]
I noticed that Cursive started not picking up deps and paths from aliases that use `:replace-deps`/`paths` and `:deps`/`paths`.

Here's a repro case using the https://github.com/imrekoszo/try-deps-tool-replace/tree/413a2fb95c3aa63eca0ba38fa073ecdb77f30864 repo.

![cursive-replace](https://user-images.githubusercontent.com/1116295/166424372-a709270e-b0e4-474e-a965-d67a5825ecf0.png)

I have all the aliases selected in the image and have refreshed the deps project after selecting them. The problems are:

1. `data.csv` not showing up as a dependency despite being in the selected `:replace` alias
2. same for `data.priority-map` in the `:nothing` alias
3. same for the paths defined in those aliases, neither `replace` nor `nothing` show up as a source root
4. `data.json` should _not_ show up as a dependency as it's in the root `:deps` and should be overridden by both `:replace-deps` in `:replace` and `:deps` in `:nothing`
5. same for the `default` directory, it should not show up as a source root but should be replaced by `nothing` and `replace`

Contrast this with the examples in the readme which show how `clojure` itself treats the combinations of those:

```shell
; clojure -X:common:extra:replace

|   :name | :paths | :deps |
|---------+--------+-------|
| default |        |       |
| replace |   true |  true |
|   extra |   true |  true |
| nothing |        |       |
|    tool |        |       |

; clojure -X:common:extra:nothing

|   :name | :paths | :deps |
|---------+--------+-------|
| default |        |       |
| replace |        |       |
|   extra |   true |  true |
| nothing |   true |  true |
|    tool |        |       |

; clojure -X:common:extra:nothing:replace

|   :name | :paths | :deps |
|---------+--------+-------|
| default |        |       |
| replace |   true |  true |
|   extra |   true |  true |
| nothing |   true |  true |
|    tool |        |       |
```

Relevant points from the readme of the linked repo:

> 1. :deps/:paths work just like :replace-deps/:replace-paths in all contexts, just as Alex Miller said in the [thread on Clojurians](https://clojurians.slack.com/archives/C6QH853H8/p1647626564105159)
> 2. :replace-* only gets rid of stuff defined at the top, not in other aliases that are used. Even when combining aliases of which some use :replace-* (or :deps/:paths, see 1.), all deps/paths from those aliases will be available.

Environment:

```
IntelliJ IDEA 2022.1 (Community Edition)
Build #IC-221.5080.210, built on April 12, 2022
Runtime version: 11.0.14.1+1-b2043.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.3.1
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 8
Registry:
    scala.erase.compiler.process.jdk.once=false

Non-Bundled Plugins:
    org.intellij.plugins.hcl (221.5080.126)
    net.seesharpsoft.intellij.plugins.csv (2.18.2)
    name.kropp.intellij.makefile (221.5080.126)
    de.femtopedia.diffplugin (2.1.7)
    com.shetline.ligatures-limited (1.0.4)
    com.jetbrains.plugins.ini4idea (221.5080.126)
    org.jetbrains.idea.grammar (2021.1.2)
    gherkin (221.5080.126)
    dev.monogon.cuelang (0.8.4)
    org.asciidoctor.intellij.asciidoc (0.37.17)
    StringToolsPlugin (4.0)
    String Manipulation (9.4.0)
    Key Promoter X (2022.1.1)
    mobi.hsz.idea.gitignore (4.4.0)
    com.cursiveclojure.cursive (1.12.3-2022.1)
    com.github.shinichy.integrant (0.1.2-imrekoszo)
    com.github.brcosta.cljstuffplugin (0.7.0)
    EDN-JSON Converter (0.0.5)
    org.intellij.scala (2022.1.13)

Kotlin: 221-1.6.20-release-285-IJ5080.210
```

**Comment by mortenschioler** [comment]
+1. I would also like to point out that the interface for alias selection is insufficiently expressive, since order matters when multiple aliases are selected. My guess is that this circumstance is at the heart of the bug.

**Comment by cursive-ide** [comment]
@mortenschioler Yes, I am planning to fix this in combination with #2409, I also think these problems are probably related.
